### DISCUSSION TITLE
[Bug Report] Stash suddenly unable to locate config.yml

### DISCUSSION THREAD

**Discussion by distkid** [stashapp/stash#5122] [Q&A] [answered]
Up to about 4 days ago stash was working perfectly, now when i start it, it says "missing the following mandatory settings: generated", did some digging and i found the config.yml in the install folder, however it was blank under the 5th line, so i tried specifying the stash folder i had "StashContent", containing the folders "screenshots" & "vtt". Still, no luck, i open up stash and my library is empty, however my stash-go.sqlite and "StashContent" folders remain full, i have spent a long time on sorting my database with stash and i'd really like to find a way to recover my database into stash! I'm not even sure what changed, but any help would be greatly appreciated!

**Comment by DogmaDragon** [comment]
1. Find your `config.yml` file. Usually located at `%userprofile%/.stash` or `$HOME/.stash`.
2. Open it and find `generated:` line. 
3. Replace with or add `generated: absolute_path_to_your_generated_folder`.
4. Save and try to start Stash.

**Reply by distkid** [reply]
> 1. Find your `config.yml` file. Usually located at `%userprofile%/.stash` or `$HOME/.stash`.
> 2. Open it and find `generated:` line.
> 3. Replace with or add `generated: absolute_path_to_your_generated_folder`.
> 4. Save and try to start Stash.

I checked C:\Users\user\.stash, and there was just a "plugins" folder. Copying the config.yml with the changes has no effect i'm afraid.
"absolute_path_to_your_generated_folder" would you be so kind as to include an example of the txt with the exact phrasing, as well as a description of how the targeted folder looks (i.e, it's contents), so i know i'm 100% specifying the correct folder?

I'll follow your instructions and let you know the result,

Thanks

**Reply by DogmaDragon** [reply]
By default, the generated folder is stored in `%userprofile%/.stash` or `$HOME/.stash`. But in your case, it sounds like you store it as "had "StashContent", containing the folders "screenshots" & "vtt"". 

So the config line should look something like `generated: full/path/to/StashContent/generated`.

**Comment by distkid** [comment]
in the root folder, i have "stash-go.sqlite", "stash-win.exe" and a folder named "Stash"
in the folder named "Stash", i have this;
![1](https://github.com/user-attachments/assets/0c5c9197-842c-4f23-9f2c-ffcfc3b1233f)

and in the folder "StashContent", i have this;
![2](https://github.com/user-attachments/assets/1595147d-d9a4-4bc2-90b3-b157cc6fdbf5)

the ".stash" folder under my user account (windows) was empty, bar one "plugins" folder, so i'm thinking i should keep it that way. Somewhere along the line some files got moved around or deleted somehow, i'm wondering if i can somehow restore what i had, given that my "Stash" folders seem to be intact with Blobs & Database files, etc.

**Reply by distkid** [reply]
Bump! Anyone who can help? :/

**Reply by WithoutPants** [reply, marked as answer]
I'm not really sure what's going on here. It looks like you've got multiple installs going on. `StashDatabase`, `StashBlobs`, `StashContent`, etc are not standard folder names that the application uses. 

The standard expected path structure should look like:
- `stash-go.sqlite`
- `config.yml`
- `blobs/`
- `cache/`
- `generated/`
- `plugins/`
- `scrapers/`

`markers`, `screenshots`, `thumbnails`, `transcodes`, `vtt` are all subdirectories that belong in the `generated` folder, so there's definitely a mix of stuff going on in those screenshots.

I'm guessing that the correct database file is the one in the root folder.

I would do the following:
- ensure `$USER/.stash` is empty. This is important because stash will look for an install here if it doesn't find a `config.yml` file in the current directory that stash was run from.
- backup that directory structure you showed screenshots of
- choose an empty directory somewhere outside of your current paths, copy (one of) your database file(s) into that directory and ensure its named `stash-go.sqlite`. 
- run stash from that directory. You will need to go through the setup process again, but it should pick up your existing database. 

That should let you determine which database is the correct one. You can then look at restoring the other folders to their correct locations. The `generated` ones aren't a huge deal as you should be able to regenerate them from your content. `blobs` is a little more difficult since that's where your cover images, performer/studio images etc are. I assume these are `StashBlobs` so you should be able to copy the contents into the `blobs` directory in your new filesystem structure.

You can always experiment with different install directories if you have multiple database files to try. Make sure that wherever you put your install is separate to where your media is.
